course_content/notebooks/dask_intro.ipynb

```diff
@@ -0,0 +1,216 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# An Introduction to Dask\n",
+    "\n",
+    "Dask is a flexible parallel computing library for Python. It allows dynamic scheduling of tasks for optimising computations, and includes collections such as arrays and lists for parallel data processing."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Terminology\n",
+    "\n",
+    "### General"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Thread\n",
+    "A thread is the simplest unit of computation.\n",
+    "\n",
+    "#### Process\n",
+    "A process is an isolated computation which consists of one or more threads."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Dask\n",
+    "\n",
+    "#### Dask architecture\n",
+    "Below is the basic structure of how the components of dask interact:\n",
+    "\n",
+    "![](../images/dask_architecture.png)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Client\n",
+    "A client is a Python class that provides the primary point of access to a scheduler and its associated workers. "
```

Comment:
The client is only used in distributed! It provides the interface between the scheduler and the users. Otherwise, this is all accurate, if a bit too general. I think I'd move the client after the section on workers, as I think it is the least important element in the dask network simply because it is not always present.

Reply:
How about:
"A client provides the primary point of access to a distributed scheduler and its associated workers. When using a threaded or multiprocessing scheduler interation with the scheduler is handled by the collection or delayed object."